### PR TITLE
Add nixos test for radicale

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change

I attempted to run radicale on Python 3 and ran into some errors along the way. This PR has a few changes in separated commits:
- a new NixOS test for radicale
- fixes to enable building radicale on Python 3
- fixes for bcrypt
- package update for passlib

cc radicale maintainers @pSub @edwtjo 

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
cryptography is updated so likely this is a mass rebuild

**Comment by aneeshusa** [comment]
cryptography doesn't need to updated for radicale so I'll split that into a different PR

File: pkgs/servers/radicale/default.nix [hunk 1]
```diff
@@ -9,13 +9,7 @@ pythonPackages.buildPythonApplication rec {
     sha256 = "1c5lv8qca21mndkx350wxv34qypqh6gb4rhzms4anr642clq3jg2";
   };
 
-  propagatedBuildInputs = [
-    pythonPackages.flup
-    pythonPackages.ldap
-    pythonPackages.sqlalchemy
-  ];
```
Comment (regnat) [inline pkgs/servers/radicale/default.nix]: You probably should keep them for python2 in order not to break existing configs

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -8622,7 +8622,7 @@ in {
       sha256 = "1z27wdxs5rj5xhhqfzvzn3yg682irkxw6dcs5jj7mcf97psk8gd8";
     };
 
-    buildInputs = with self; [ nose pybcrypt];
+    propagatedBuildInputs = with self; [ nose bcrypt ];
```
Comment (regnat) [inline pkgs/top-level/python-packages.nix]: Is there a reason why you moved it from `buildInputs` to `propagatedBuildInputs` ?

Comment (grahamc) [inline pkgs/top-level/python-packages.nix]: bump @aneeshusa

File: pkgs/servers/radicale/default.nix (hunk 1 above)
Comment (aneeshusa) [inline pkgs/servers/radicale/default.nix]: Done.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (aneeshusa) [inline pkgs/top-level/python-packages.nix]: Yeah, `bcrypt` needs to be propagated for modules that depend on `passlib` to be able to use the `bcrypt` backend. I added this to the commit message.

**Comment by aneeshusa** [comment]
Thanks for taking a look, I've addressed review comments and re-tested this. I also added a commit adding myself as a maintainer.

**Comment by aneeshusa** [comment]
I've rebased this on master. I'd like to have this merged before Radicale 2.0 comes out of beta if someone wants to take a look!

**Pull request merged**